File: jtaf-spring/src/main/java/ch/jtaf/security/WebSecurityConfig.java
package ch.jtaf.security;

import org.springframework.context.annotation.Configuration;
import org.springframework.http.HttpMethod;
import org.springframework.security.config.annotation.web.builders.HttpSecurity;
import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;

@Configuration
@EnableWebSecurity
public class WebSecurityConfig extends WebSecurityConfigurerAdapter {

    @Override
    protected void configure(HttpSecurity http) throws Exception {
        http
                .authorizeRequests()
                .antMatchers("/", "/css/**", "/i18n/**", "/images/**", "/util/**").permitAll()
                .antMatchers("/input/**").hasRole("user")
                .antMatchers("/spaces/**").hasRole("user")
                .antMatchers("/user/**").hasRole("user")
                .antMatchers(HttpMethod.PUT, "/res/users/**").hasRole("user")
                .antMatchers(HttpMethod.DELETE, "/res/users/**").hasRole("user")
                .antMatchers(HttpMethod.POST, "/res/athletes/**", "/res/categories/**", "/res/clubs/**", "/res/competitions/**", "/res/events/**", "/res/series/**", "/res/spaces/**").hasRole("user")
                .antMatchers(HttpMethod.PUT, "/res/athletes/**", "/res/categories/**", "/res/clubs/**", "/res/competitions/**", "/res/events/**", "/res/series/**", "/res/spaces/**").hasRole("user")
                .antMatchers(HttpMethod.DELETE, "/res/athletes/**", "/res/categories/**", "/res/clubs/**", "/res/competitions/**", "/res/events/**", "/res/series/**", "/res/spaces/**").hasRole("user")
                .and()
                .httpBasic();
    }
}

File: src/main/kotlin/ch/jtaf/config/WebSecurityConfig.kt
package ch.jtaf.config

import ch.jtaf.control.service.JtafUserDetailsService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder
import org.springframework.security.config.annotation.web.builders.HttpSecurity
import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity
import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder
import org.springframework.security.crypto.password.PasswordEncoder

@Configuration
@EnableWebSecurity
class WebSecurityConfig : WebSecurityConfigurerAdapter() {

    override fun configure(http: HttpSecurity) {
        http
                .authorizeRequests()
                .antMatchers("/actuator/health").permitAll()
                .antMatchers("/actuator/**", "/sec/**").authenticated()
                .anyRequest().permitAll()
                .and().formLogin().loginPage("/login").permitAll()
                .and().logout().logoutUrl("/logout").logoutSuccessUrl("/dashboard").permitAll()
    }

    @Autowired
    fun configureGlobal(auth: AuthenticationManagerBuilder, jtafUserDetailsService: JtafUserDetailsService) {
        auth.userDetailsService(jtafUserDetailsService).passwordEncoder(passwordEncoder())
    }

    @Bean
    fun passwordEncoder(): PasswordEncoder {
        return BCryptPasswordEncoder()
    }
}

